File: 7KYU/Count the Digit/solution.c
int getCount(int sq, int d) {
  
  int count = 0;
  if(sq == d) {
    return 1;
  }
  
  while(sq) {
    if(sq % 10 == d) {
      count++;
    }
    sq /= 10;
  }
  return count;
}

int nbDig(int n, int d) {
  int count = 0;
  
  for(int i = 0; i <= n; i++) {
    int sq = i * i;
    count += getCount(sq, d);
  }
  
  return count;
}

File: 7KYU/Count the Digit/solution.go
package kata

func getDigitCount(n int, d int) int {
	if n == d {
		return 1
	}
	var count int = 0

	for {
		if n%10 == d {
			count++
		}
		n = n / 10
		if n < 1 {
			break
		}
	}

	return count
}

func NbDig(n int, d int) int {
	var count int = 0

	for i := 0; i <= n; i++ {
		count = count + getDigitCount(i*i, d)
	}

	return count
}
